var/spack/repos/builtin/packages/tcsh/package.py

```diff
@@ -0,0 +1,62 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Tcsh(AutotoolsPackage):
+    """
+    Tcsh is an enhanced but completely compatible version of csh, the C shell.
+    Tcsh is a command language interpreter which can be used both as an
+    interactive login shell and as a shell script command processor. Tcsh
+    includes a command line editor, programmable word completion, spelling
+    correction, a history mechanism, job control and a C language like syntax.
+    """
+
+    homepage = "http://www.tcsh.org/"
+    url      = "ftp://ftp.astron.com/pub/tcsh/tcsh-6.20.00.tar.gz"
+
+    version('6.20.00', '59d40ef40a68e790d95e182069431834')
+
+    # Upstream patches
+    patch('tcsh-6.20.00-000-add-all-flags-for-gethost-build.patch')
+    patch('tcsh-6.20.00-001-delay-arginp-interpreting.patch')
+    patch('tcsh-6.20.00-002-type-of-read-in-prompt-confirm.patch')
+    patch('tcsh-6.20.00-003-fix-out-of-bounds-read.patch')
+    patch('tcsh-6.20.00-004-do-not-use-old-pointer-tricks.patch')
+    patch('tcsh-6.20.00-005-reset-fixes-numbering.patch')
+    patch('tcsh-6.20.00-006-cleanup-in-readme-files.patch')
+    patch('tcsh-6.20.00-007-look-for-tgetent-in-libtinfo.patch')
+    patch('tcsh-6.20.00-008-guard-ascii-only-reversion.patch')
+    patch('tcsh-6.20.00-009-fix-regexp-for-backlash-quoting-tests.patch')
+
+    # Downstream patches
+    patch('tcsh-6.20.00-manpage-memoryuse.patch')
```

Comment:
You're probably going to want to add `when='@6.20.00'` to all of these. As soon as someone adds a new version, it will break.